hypervisor/src/kvm/mod.rs

```diff
@@ -250,17 +258,21 @@ impl vm::Vm for KvmVm {
         readonly: bool,
         log_dirty_pages: bool,
     ) -> MemoryRegion {
+        if log_dirty_pages {
+            self.dirty_log_slots.write().unwrap().push(KvmDirtyLogSlot {
+                slot,
+                guest_phys_addr,
+                memory_size,
+                userspace_addr,
+            });
+        }
+
```

Comment:
This seems wrong to me. This function is used to make the MemoryRegion(both KVM and MSHV)  parameter , this is not the setting the memory regions. We should populate the slots in the create_user_memory_region call and remove the slot in the remove_user_memory_region function.

Reply:
Good point. Will update the PR shortly. 

If there is anything else missing (particularly for you to integrate with MSHV), please let me know. Thank you.

Reply:
Otherwise seems good to me.

Reply:
Cool. Updated. PTAL.